dns/dyndns/src/www/services_dyndns_edit.php

```diff
@@ -153,7 +155,8 @@ function is_dyndns_username($uname)
         $dyndns['interface'] = $pconfig['interface'];
         $dyndns['zoneid'] = $pconfig['zoneid'];
         $dyndns['resourceid'] = $pconfig['resourceid'];
-        $dyndns['ttl'] = $pconfig['ttl'];
+		//This field *must* finally be an integer. So anything else has to be filtered...
+        $dyndns['ttl'] = intval(preg_replace('/[^0-9\.]/', '', $pconfig['ttl']));
```

Comment:
would be more appropriate to add a validator instead of scrubbing input. casting gets los along the way worst case... data type expected by the other side must be set correctly in the request

Reply:
@andreas-rupper you need help with the validation? I see this is marked resolved, but I don't see other code.

Reply:
@AdSchellevis 
I was thinking about the validation, But I'm not sure how to do it. Probably with some JavaScript in the browser? Or with some other php based validation? This might bloat the code...
If you want this to be simple, then I can revert it to the original state.
At least my code filters anything (including any typos such as space) that that is not an integer even from within a string. i.e. if someone wanted the value "120" and by some error typed "12 0" this would return 120.
And TTL must be integer by its nature...
What do you think?

Reply:
@andreas-rupper ah, I see, I already thought it wasn't clear, hence the question :) Validation in the legacy code isn't very complex, usually you just add an item to $input_errors, like:

https://github.com/opnsense/plugins/blob/985617674762a1ff90cca67c08a44ef019dbc9b1/dns/dyndns/src/www/services_dyndns_edit.php#L135-L137

In your case this would be something like:
```
if ((string)((int)$pconfig['ttl']) != $pconfig['ttl']) {
   $input_errors[] = gettext("The TTL value needs to be a valid integer number.");
}
```

Which prevents invalid data from being saved via the form.